doc/staging.md

```diff
@@ -0,0 +1,75 @@
+tool is run.
+
+Things to note:
+
+* Pull requests are only merged from users that are on a trusted list, stored
+  in the `kernelci-deploy` configuration files.
+* Pull requests are merged in order of their number, so older ones take
+  precedence.
+* Pull requests that fail to merge are ignored and will not be tested.
+* If any of the patches stored in `kernelci-deploy` doesn't apply, the
+  resulting branch is not pushed.  All the patches must apply, since some of
+  them are necessary to adjust the staging behaviour (say, to not send
+  bisection email reports).
+* A tag is created with the current date and pushed with the branch.
+
+
+### Jenkins: bot.staging.kernelci.org
+
+The staging instance is running Jenkins, just like production.  The main
+difference is that the staging one is publicly visible, read-only for anonymous
+users: [bot.staging.kernelci.org](https://bot.staging.kernelci.org/)
+
+This allows for the job logs to be inspected.  Also, some developers have a
+personal folder there to run modified versions of the Jenkins job but still
+using the available resources (builders, API tokens to submit jobs in test
+labs...).
+
+
+### Run every 8h
+
+There is a timer on the staging.kernelci.org server which starts a job every
+8h, so 3 times per day.  The job does the following:
+
+1. update [staging branch for `kernelci-jenkins`](https://github.com/kernelci/kernelci-jenkins/tree/staging.kernelci.org)
+1. recreate Jenkins jobs by running the job-dsl "seed" job
+1. update [staging branch for `kernelci-core`](https://github.com/kernelci/kernelci-core/tree/staging.kernelci.org)
+1. update [staging branch for `kernelci-backend`](https://github.com/kernelci/kernelci-backend/tree/staging.kernelci.org)
+1. update the `kernelci-backend` service using Ansible from [`kernelci-backend-config`](https://github.com/kernelci/kernelci-backend-config) with the staging branch
+1. update [staging branch for `kernelci-frontend`](https://github.com/kernelci/kernelci-frontend/tree/staging.kernelci.org)
+1. update the `kernelci-frontend` service using Ansible from [`kernelci-frontend-config`](https://github.com/kernelci/kernelci-frontend-config) with the staging branch
+1. create and push a `staging.kernelci.org` branch with a tag to the [kernelci Linux kernel repo](https://github.com/kernelci/linux)
+1. trigger a monitor job in Jenkins with the [`kernelci_staging`](https://github.com/kernelci/kernelci-core/blob/staging.kernelci.org/build-configs.yaml#L612) config
+
+The last step should cause the monitor job to detect that the staging kernel
+branch has been updated, and run a kernel build trigger job which in turn will
+cause tests to be run.  Builds and test results will be sent to the staging
+backend instance, and results will be available on the staging web dashboard.
+Regressions will cause bisections to be run on the staging instance, and
+results to be sent to the `kernelci-results-staging@groups.io` mailing list.
```

Comment:
You could link the email address here to the groups.io page so people know where to sign up.

Reply:
Done.

Reply:
LGTM